Assignment3.java

```diff
@@ -0,0 +1,200 @@
+import java.util.*;
+public class Assignment3 {
+	
+	private final static String right="Ok";
+	private final static String wrong="not Ok";
+	
+	
+
+private static <T> String convertTestResultToString(T result, T expectedResult){
+	if (result==null){
+		if (expectedResult==null){
+			return right;
+		}
+		else
+			return wrong;
+	}
+	if (result.equals(expectedResult))
+		return right;
+	return wrong;
+}
+
+public static  void test1(){
+	String [] w ={"car","card","cart","cat"};
+	List <String> words=Arrays.asList(w);
+	ArrayList <String> prefics =new ArrayList(words);
+	prefics.add("c");
+	prefics.add("ca");
+	Dictionary dictionary=new Dictionary(words, prefics);
+	char[][] grid={{'a','a','r'},{'t','c','d'}};
+	ForGridWalker forGridWalker=new ForGridWalker(grid,dictionary);
+	forGridWalker.start();
+	TreeSet <String> answer=new TreeSet();
+	answer.add("cat"); answer.add("card"); answer.add("car");
+	System.out.println(convertTestResultToString(forGridWalker.finish(), answer));
+}
+public static void test2(){
+	String [] w ={"word","dor","or","wdw"};
+	List <String> words=Arrays.asList(w);
+	ArrayList <String> prefics =new ArrayList(words);
+	prefics.add("w");
+	prefics.add("o");
+	prefics.add("wo");
+	prefics.add("wor");
+	prefics.add("d");
+	prefics.add("do");
+	prefics.add("wd");
+	Dictionary dictionary=new Dictionary(words, prefics);
+	char[][] grid={{'o','d'},{'r','w'}};
+	ForGridWalker forGridWalker=new ForGridWalker(grid,dictionary);
+	forGridWalker.start();
+	TreeSet <String> answer=new TreeSet();
+	answer.add("word"); answer.add("dor"); answer.add("or");
+	//System.out.println(forGridWalker.finish());
+	System.out.println(convertTestResultToString(forGridWalker.finish(), answer));
+}
+public static void test3(){
+	String [] w ={"abba","abet","faab","kgcbtk","cab"};
+	List <String> words=Arrays.asList(w);
+	ArrayList <String> prefics =new ArrayList(words);
+	prefics.add("a");
+	prefics.add("ab");
+	prefics.add("abb");
+	prefics.add("abe");
+	prefics.add("f");
+	prefics.add("fa");
+	prefics.add("faa");
+	prefics.add("k");
+	prefics.add("kg");
+	prefics.add("kgc");
+	prefics.add("kgcb");
+	prefics.add("kgcbt");
+	prefics.add("c");
+	prefics.add("ca");
+	Dictionary dictionary=new Dictionary(words, prefics);
+	char[][] grid={{'a','b','c'},{'a','f', 'a'},{'g','c','b'},{'k','t','e'}};
+	ForGridWalker forGridWalker=new ForGridWalker(grid,dictionary);
+	forGridWalker.start();
+	TreeSet <String> answer=new TreeSet();
+	answer.add("abet"); answer.add("faab"); answer.add("cab");
+	//System.out.println(forGridWalker.finish());
+	System.out.println(convertTestResultToString(forGridWalker.finish(), answer));
+}
+public static void test(){
```

Comment:
The tests look good. You could also look into JUnit, it's a Java framework for writing tests and has a lot of cool features. 
Also, I think that the tests should belong in another class - GridWalkerTests, for example.